### PR TITLE
Feature: Train and run an XGBoost model

### PR DESCRIPTION
**Pull request by ElectricMountains**
#### Reference Issues/PRs

This PR is related to the Roadmap of Q1 #6 + Roadmap of Q2 #21

#### What does this implement/fix? Explain your changes.

This PR introduces the XGBoost model that uses the 2019 toktarova data along with our collected weather data to replicate the GEGIS approach.

#### Any other comments?

I've exported the jupyter notebook to a python file to facilitate an easier review process.

### CODE REVIEW

**Comment by ElectricMountains** [comment]
Integrated the main points of the feedback from @eantonini, namely:
- Normalized the hourly demand
- Created a visualization to compare the ground truth with the predicted for the cross-validation.

**Review by eantonini — COMMENTED** [review]
Thanks for developing this nice ML model and getting it up and running.

I have two suggestions:
- First, it would be helpful to unpack the Jupyter notebook into a few separate Python scripts:
  - one script for pre-processing the input data,
  - one script for setting up and running the ML model,
  - and one script for analyzing the results.
- Second, could you generate a figure that combines some of the plots you already have, so we can include it in our slides? I’m thinking of a 3-panel figure: on the left, a large panel showing the mean absolute errors; and on the right, two smaller panels showing the historical and predicted time series for the best- and worst-performing countries.
